src/fromager/commands/bootstrap.py

```diff
@@ -402,3 +405,118 @@ def write_constraints_file(
 
 
 bootstrap._fromager_show_build_settings = True  # type: ignore
+
+
+@click.command()
+@click.option(
+    "-r",
+    "--requirements-file",
+    "requirements_files",
+    multiple=True,
+    type=str,
+    help="pip requirements file",
+)
+@click.option(
+    "-p",
+    "--previous-bootstrap-file",
+    "previous_bootstrap_file",
+    type=str,
+    help="graph file produced from a previous bootstrap",
+)
+@click.option(
+    "-c",
+    "--cache-wheel-server-url",
+    "cache_wheel_server_url",
+    help="url to a wheel server from where fromager can download the wheels that it has built before",
+)
+@click.option(
+    "--skip-constraints",
+    "skip_constraints",
+    is_flag=True,
+    default=False,
+    help="Skip generating constraints.txt file to allow building collections with conflicting versions",
+)
+@click.option(
+    "-f",
+    "--force",
+    is_flag=True,
+    default=False,
+    help="rebuild wheels even if they have already been built",
+)
+@click.option(
+    "-m",
+    "--max-workers",
+    type=int,
+    default=None,
+    help="maximum number of parallel workers to run (default: unlimited)",
+)
+@click.argument("toplevel", nargs=-1)
+@click.pass_obj
+@click.pass_context
+def bootstrap_parallel(
+    ctx: click.Context,
+    wkctx: context.WorkContext,
+    *,
+    requirements_files: list[str],
+    previous_bootstrap_file: str | None,
+    cache_wheel_server_url: str | None,
+    skip_constraints: bool,
+    force: bool,
+    max_workers: int | None,
+    toplevel: list[str],
+) -> None:
+    """Bootstrap and build-parallel
+
+    Bootstraps all dependencies in sdist-only mode, then builds the
+    remaining wheels in parallel. The bootstrap step downloads sdists
+    and builds build-time dependency in serial. The build-parallel step
+    builds the remaining wheels in parallel.
+    """
+    # Do not remove build environments in bootstrap phase to speed up the
+    # parallel build phase.
+    logger.info("keep build env for build-parallel phase")
+    wkctx.cleanup_buildenv = False
+
+    start = time.perf_counter()
+    logger.info("*** starting bootstrap in sdist-only mode ***")
+    ctx.invoke(
+        bootstrap,
+        requirements_files=requirements_files,
+        previous_bootstrap_file=previous_bootstrap_file,
+        cache_wheel_server_url=cache_wheel_server_url,
+        sdist_only=True,
+        skip_constraints=skip_constraints,
+        toplevel=toplevel,
+    )
+
+    # statistics
+    wheels = sorted(f.name for f in wkctx.wheels_downloads.glob("*.whl"))
+    sdists = sorted(f.name for f in wkctx.sdists_downloads.glob("*.tar.gz"))
+    logger.debug("wheels: %s", ", ".join(wheels))
```

Comment:
These log lines could end up being pretty massive for our full index builds. Let's see how it goes, but I anticipate coming back and either listing a few packages per line or removing the log calls entirely.